api/server/request_controller.go

```diff
@@ -0,0 +1,117 @@
+package server
+
+import (
+	"net/http"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/gin-gonic/gin"
+	"github.com/iron-io/functions/api"
+	"github.com/iron-io/runner/common"
+)
+
+type RequestController interface {
+	AppName() string
```

Comment:
@treeder If you follow the request flow there's no App (unless its name) and Route (unless its path)  information until the request reaches the final request handler. So from where would, that information, come from? As far as I know, there's no database before the final handler, maybe inside middlewares. Are you thinking about middlewares doing those database calls, and writing that App/Route struct so in the final handler, instead of calling the database again, it would just use the struct?

Sorry I'm not understanding the need of a App/Route struct in this situation